treewide: boards: use `zephyr:board-supported-hw` in most boards' documentation

mathieuchopstm opened the pull request:
Split in one commit per board vendor to ease review but I can squash.

Replaces hand-written tables with the `zephyr:board-supported-hw` directive. In most cases, this is fine - however, some dual-SoC/etc platforms have multiple pages which doesn't work with `zephyr:board` - these have been left unmodified.

While at it, `.. zephyr:board::` was added to some boards on which it was missing, and old `:ref:`s are updated to use `:zephyr:board:` references instead.

Tables were found using regexp: `\|\s*Interface\s*\|\s*Controller\s*\|\s*Driver/Component\s*\|`

All should be covered now:
```
~/zephyrproject/zephyr$ grep -r ".. zephyr:board-supported-hw::" --include="*.rst" boards/ | wc -l
805
~/zephyrproject/zephyr$ grep -r "Supported Features" --include="*.rst" boards/ | wc -l
804
```
(one file has two `.. zephyr:board-supported-hw::` directives... might not be ideal)

mathieuchopstm committed:
boards: actinius: use zephyr:board-supported-hw

Replace legacy hand-mantained tables with the new directive that generates
a table automatically based on platform device tree.

Signed-off-by: Mathieu Choplain <mathieu.choplain-ext@st.com>

mathieuchopstm committed:
boards: adafruit: use zephyr:board-supported-hw

Replace legacy hand-mantained tables with the new directive that generates
a table automatically based on platform device tree.

Signed-off-by: Mathieu Choplain <mathieu.choplain-ext@st.com>

mathieuchopstm committed:
boards: beagle: use zephyr:board-supported-hw

Replace legacy hand-mantained tables with the new directive that generates
a table automatically based on platform device tree.

Signed-off-by: Mathieu Choplain <mathieu.choplain-ext@st.com>

mathieuchopstm committed:
boards: cypress: use zephyr:board-supported-hw

Replace legacy hand-mantained tables with the new directive that generates
a table automatically based on platform device tree.

Signed-off-by: Mathieu Choplain <mathieu.choplain-ext@st.com>

mathieuchopstm committed:
boards: lowrisc: use zephyr:board-supported-hw

Replace legacy hand-mantained tables with the new directive that generates
a table automatically based on platform device tree.

Signed-off-by: Mathieu Choplain <mathieu.choplain-ext@st.com>

mathieuchopstm committed:
boards: qemu: use zephyr:board-supported-hw

Replace legacy hand-mantained tables with the new directive that generates
a table automatically based on platform device tree.

Signed-off-by: Mathieu Choplain <mathieu.choplain-ext@st.com>

mathieuchopstm committed:
boards: toradex: use zephyr:board-supported-hw

Replace legacy hand-mantained tables with the new directive that generates
a table automatically based on platform device tree.

Signed-off-by: Mathieu Choplain <mathieu.choplain-ext@st.com>

wearyzen commented on boards/arm/mps3/doc/index.rst:
```diff
@@ -224,31 +224,7 @@ ARM MPS3 provides the following hardware components:
-+-----------+------------+-------------------------------------+
-
-Other hardware features are not currently supported by the port.
-See the `MPS3 FPGA Website`_ for a complete list of MPS3 AN547 board hardware
```
Thank you for the change!
Overall for Arm boards, removing the "default configuration can be found in" is ok but I would like to keep the statements "Other hardware features are not currently supported by the port.
See the `MPS3 FPGA Website`_ for a complete list of MPS3 AN547 board hardware" and the link to the fpga boards.

jadonk reviewed (dismissed):
Looks good for boards/beagle.

wearyzen commented:
Hi @mathieuchopstm, do you plan to work on this?

bjarki-andreasen reviewed (dismissed):
Neat

mathieuchopstm commented:
Rebased to resolve conflict due to #101580 merge

mathieuchopstm committed:
boards: waveshare: use zephyr:board-supported-hw

Replace legacy hand-mantained tables with the new directive that generates
a table automatically based on platform device tree.

Signed-off-by: Mathieu Choplain <mathieu.choplain-ext@st.com>

mathieuchopstm committed:
boards: snps: use zephyr:board-supported-hw

Replace legacy hand-mantained tables with the new directive that generates
a table automatically based on platform device tree.

Signed-off-by: Mathieu Choplain <mathieu.choplain-ext@st.com>

mathieuchopstm committed:
boards: vngiotlab: use zephyr:board-supported-hw

Replace legacy hand-mantained tables with the new directive that generates
a table automatically based on platform device tree.

Signed-off-by: Mathieu Choplain <mathieu.choplain-ext@st.com>

mathieuchopstm committed:
boards: 96boards: use zephyr:board-supported-hw

Replace legacy hand-mantained tables with the new directive that generates
a table automatically based on platform device tree. While at it, in some
boards, the existing "markers" have been replaced with the zephyr:board
role as it is required for proper operation of zephyr:board-supported-hw.
Some documentation pages which contained a reference to the marker are also
updated to use zephyr:board instead.

Signed-off-by: Mathieu Choplain <mathieu.choplain-ext@st.com>

mathieuchopstm committed:
boards: brcm: use zephyr:board-supported-hw

Replace legacy hand-mantained tables with the new directive that generates
a table automatically based on platform device tree. While at it, in some
boards, the existing "markers" have been replaced with the zephyr:board
role as it is required for proper operation of zephyr:board-supported-hw.

Signed-off-by: Mathieu Choplain <mathieu.choplain-ext@st.com>

mathieuchopstm committed:
boards: digilent: use zephyr:board-supported-hw

Replace legacy hand-mantained tables with the new directive that generates
a table automatically based on platform device tree. While at it, in some
boards, the existing "markers" have been replaced with the zephyr:board
role as it is required for proper operation of zephyr:board-supported-hw.

Signed-off-by: Mathieu Choplain <mathieu.choplain-ext@st.com>

mathieuchopstm committed:
boards: fanke: use zephyr:board-supported-hw

Replace legacy hand-mantained tables with the new directive that generates
a table automatically based on platform device tree. While at it, in some
boards, the existing "markers" have been replaced with the zephyr:board
role as it is required for proper operation of zephyr:board-supported-hw.

Signed-off-by: Mathieu Choplain <mathieu.choplain-ext@st.com>

mathieuchopstm committed:
boards: intel: use zephyr:board-supported-hw

Replace legacy hand-mantained tables with the new directive that generates
a table automatically based on platform device tree. While at it, in some
boards, the existing "markers" have been replaced with the zephyr:board
role as it is required for proper operation of zephyr:board-supported-hw.

Signed-off-by: Mathieu Choplain <mathieu.choplain-ext@st.com>

mathieuchopstm committed:
boards: mikroe: use zephyr:board-supported-hw

Replace legacy hand-mantained tables with the new directive that generates
a table automatically based on platform device tree. While at it, in some
boards, the existing "markers" have been replaced with the zephyr:board
role as it is required for proper operation of zephyr:board-supported-hw.
Some documentation pages which contained a reference to the marker are also
updated to use zephyr:board instead.

Signed-off-by: Mathieu Choplain <mathieu.choplain-ext@st.com>

mathieuchopstm committed:
boards: nordic: use zephyr:board-supported-hw

Replace legacy hand-mantained tables with the new directive that generates
a table automatically based on platform device tree. While at it, in some
boards, the existing "markers" have been replaced with the zephyr:board
role as it is required for proper operation of zephyr:board-supported-hw.
Some documentation pages which contained a reference to the marker are also
updated to use zephyr:board instead.

Signed-off-by: Mathieu Choplain <mathieu.choplain-ext@st.com>

mathieuchopstm committed:
boards: phytec: use zephyr:board-supported-hw

Replace legacy hand-mantained tables with the new directive that generates
a table automatically based on platform device tree. While at it, in some
boards, the existing "markers" have been replaced with the zephyr:board
role as it is required for proper operation of zephyr:board-supported-hw.
Some documentation pages which contained a reference to the marker are also
updated to use zephyr:board instead.

Signed-off-by: Mathieu Choplain <mathieu.choplain-ext@st.com>

mathieuchopstm committed:
boards: pjrc: use zephyr:board-supported-hw

Replace legacy hand-mantained tables with the new directive that generates
a table automatically based on platform device tree. While at it, in some
boards, the existing "markers" have been replaced with the zephyr:board
role as it is required for proper operation of zephyr:board-supported-hw.

Signed-off-by: Mathieu Choplain <mathieu.choplain-ext@st.com>

mathieuchopstm committed:
boards: sparkfun: use zephyr:board-supported-hw

Replace legacy hand-mantained tables with the new directive that generates
a table automatically based on platform device tree. While at it, in some
boards, the existing "markers" have been replaced with the zephyr:board
role as it is required for proper operation of zephyr:board-supported-hw.

Signed-off-by: Mathieu Choplain <mathieu.choplain-ext@st.com>

mathieuchopstm committed:
boards: u-blox: use zephyr:board-supported-hw

Replace legacy hand-mantained tables with the new directive that generates
a table automatically based on platform device tree. While at it, in some
boards, the existing "markers" have been replaced with the zephyr:board
role as it is required for proper operation of zephyr:board-supported-hw.

Signed-off-by: Mathieu Choplain <mathieu.choplain-ext@st.com>

mathieuchopstm committed:
boards: xen: use zephyr:board-supported-hw

Replace legacy hand-mantained tables with the new directive that generates
a table automatically based on platform device tree. While at it, in some
boards, the existing "markers" have been replaced with the zephyr:board
role as it is required for proper operation of zephyr:board-supported-hw.

Signed-off-by: Mathieu Choplain <mathieu.choplain-ext@st.com>

mathieuchopstm committed:
boards: shields: update dead references to boards

As some boards were updated to use `zephyr:board` instead of a mere marker,
old `:ref:` references have become invalid in some shields' documentation.
Fix it by using `:zephyr:board:` in these shields' docs.

Signed-off-by: Mathieu Choplain <mathieu.choplain-ext@st.com>

wearyzen commented:
The PR doesn't have any Arm documentation so removing myself from the assignee list.

mathieuchopstm committed:
samples: sensor: lsm6dsl: add reference to board using zephyr:board

Board `disco_l475_iot1` is mentioned by the sample's README in plain text;
replace with the zephyr:board role which looks prettier and links to the
actual board's documentation.

Signed-off-by: Mathieu Choplain <mathieu.choplain-ext@st.com>

kartben reviewed (approved):
Many thanks, @mathieuchopstm ! ❤️
